conformance_tests/tools/metrics/src/test_metric.cpp

```diff
@@ -1473,6 +1478,9 @@ TEST_F(
         LOG_DEBUG << "additional sleep before expecting event to be ready "
                   << timeLeft;
         std::this_thread::sleep_for(std::chrono::seconds(timeLeft));
+        // Need to keep runnning some workload since some platforms go into sleep mode when idle and dont generate reports. This will delay the event generation.
```

Comment:
We can be more subtle about platform specific behaviour in the opensource CTS tests.
One option is to have a parallel thread running which continuously runs workloads till the test conditions in the main thread are met

Reply:
Changed the comment to make it more generic. clang-format has introduced some format changes for previous PR.

> One option is to have a parallel thread running which continuously runs workloads till the test conditions in the main thread are met

can we take this up as a separate task?